pum/core/deltapy.py

```diff
@@ -7,24 +7,26 @@ class DeltaPy(metaclass=ABCMeta):
     """This abstract class must be instantiated by the delta.py classes"""
 
     def __init__(
-            self, current_db_version, delta_dir, delta_dirs, pg_service, upgrades_table, variables: dict={}):
```

Comment:
obsessed with default values ;) ?
seriously, what does it bring?

Reply:
Not particularly, but obsessed with removing unused/unneeded stuff ;-)

Here the empty dict created for the default value is never used, because the argument is always passed to the constructor by the calling code.

And why having a default value for `variables` and not for the other arguments? What makes `variables` special? This is confusing.

And also, it's often a bad idea to use a mutable object as an argument default value, because that object will be shared across all the function calls. In our case it will be shared by all the DeltaPy instances, which is source of confusion and bugs.